### ISSUE TITLE
Uniform API for key bindings

### ISSUE THREAD

**Issue by tommoor** [nwjs/nw.js#2968]
I noticed this week that the API for key bindings in MenuItem's and global shortcuts are different, this is unnecessary and confusing and they should be brought inline. For example, to achieve the same thing here:

In global shortcuts:
```js
new gui.Shortcut({
  key : "Ctrl+Alt+I",
  active: function() {
  }
});
```

In a menu item:

```js
new gui.MenuItem({
    label: 'My Label',
    key: "i",
    modifiers: "cmd-alt",
    click: function() {
    }
  })
```

**Comment by mrfabbri** [comment]
@tommoor :+1: it makes definitely sense.

- [ ] `MenuItem` support for `key: "ctrl+alt+i"` style shortcuts (this can be done in JavaScript in `MenuItem.js`) while retaining backward compatibility (`key` + `modifiers`). If modifiers are present in `key` and also `modifiers` is provided, the modifiers in `modifiers` are ignored.
- [ ] `Click`/`active` interchangeable in `MenuItem`.
- [x] `key` property is case insensitive (e.g. "ctrl+alt+i" and "CTRL+ALT+I" would be equally fine) in both `MenuItem` and `Shortcut`.
- [ ] Thorough automatic tests to assure we keep backward compatibility.
- [ ] Update manual tests.
- [ ] Update the docs (deprecate `key`+`modifiers` in `MenuItem`).

I noticed that `Shortcut` translates `ctrl` to `cmd` on the Mac OS X (and ignores `cmd`, so you can't use `ctrl` in a global shortcut on Mac OS X) while `MenuItem` supports both `ctrl` and `cmd` (it's explained in the docs so it's ok but it can be a little surprising to the user).

@rogerwang @jtg-gg @ghostoy what's your take on this?

**Comment by tommoor** [comment]
Glad you brought up the ctrl/cmd translation on Mac, I can see the thinking behind this but it's actually pretty annoying and restrictive - Mac has a ctrl AND a command key so you should be able to map shortcuts to both.

**Comment by octalmage** [comment]
I've seen another language use Mod to mean Ctrl on Windows and Command on Mac, and I like this implementation. Cmd and Ctrl will also work, but Mod will select the correct one.

**Comment by tommoor** [comment]
@octalmage neat idea, do you remember where you saw it?

**Comment by octalmage** [comment]
@tommoor here it is! 

http://craig.is/killing/mice

Look under the "Combination of keys" section.

**Comment by tommoor** [comment]
I think it's a clever solution that achieves both things well, thoughts @mrfabbri @rogerwang ?

**Comment by mrfabbri** [comment]
@otcalImage @toommoor pretty neat trick! LGTM

Now [in case this gets green light] I wonder if we should eventually adopt the same strategy for MenuItems key bindings, because having cmd bound to cmd in one place and bound to ctrl in another will be confusing.

**Comment by tommoor** [comment]
Yep, ideally - I'm trying to push or us to get some uniformity in this project not make it worse :smiley:

**Comment by mrfabbri** [comment]
:smile:

**Comment by tommoor** [comment]
Global keyboard shortcuts also need a keyUp keyDown binding, "active" isn't enough information for a lot of usecases. I should probably pull this out into it's own issue...

**Comment by EladBezalel** [comment]
I've finished working on windows hook listener and my only problem right now is that the hook requires an infinite loop to process messages, calling the listener will block the rest of the application,

I thought about running it in a separate thread/webworker but it feels wrong and i'm afraid it might not work,

any other ideas?
